enhancements/authentication/synchronizing-user-groups-on-login.md

```diff
@@ -0,0 +1,247 @@
+---
+title: synchronizing-user-groups-on-login
+authors:
+  - "@stlaz"
+reviewers:
+  - "@sur"
+  - "@slaskawi"
+approvers:
+  - "@deads2k"
+  - "@sttts"
+creation-date: 2021-05-22
+last-updated: yyyy-mm-dd
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Synchronizing User Groups on Login
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+OpenShift should be capable of synchronizing user groups from a 3rd party
+identity provider upon a user logging in to their OpenShift cluster.
+
+## Motivation
+
+In the real world, OpenShift is not the owner of user identities but these
+are stored in a 3rd party identity provider system. When OpenShift retrieves
+such an identity, it usually only gathers information about the user's unique
+identification in that IdP and optionally a username that's preferred for the
+user if it's different from its identifier.
+
+It is a common practice to group users in logical groups on the identity provider
+side and sometimes it would make sense to bring the groups over to the OpenShift
+cluster as well. The identity providers oftentimes allow a protocol-specific path
+to retrieve group information, which is something OpenShift should leverage in those
+cases where it is desired.
+
+### Goals
+
+- allow retrieving user's groups from a 3rd party OIDC identity provider into
+  OpenShift when such a user logs in
```

Comment:
doesn't this assume that all groups presented are 
1. valid for openshift
2. desired in the cluster
3. are the same for all tokens issued

?

Reply:
> doesn't this assume that all groups presented are
>
>    1. valid for openshift
>    2. desired in the cluster
 
It does. I suppose we could make lists to allow/block certain groups appear in the generic provider API.

> 3. are the same for all tokens issued
yes